### ISSUE TITLE
Use something else than id to identify RC when stopping app 

### ISSUE THREAD

**Issue by vpavlin** [projectatomic/atomicapp#147]
Currently Kubernetes provider uses `id` to identify replication controller and fails if not present. Add some other way how to get rc name/id to be able to resize to 0 (to kill the pods) on stop call

```
2015-06-11 07:48:29,227 - kubernetes - INFO - Undeploying from Kubernetes
2015-06-11 07:48:29,227 - kubernetes - DEBUG - ./.workdir/guestbookfront-app/artifacts/k8s/guestbook-controller.json
2015-06-11 07:48:29,228 - kubernetes - DEBUG - ./.workdir/guestbookfront-app/artifacts/k8s/guestbook-service.json
Traceback (most recent call last):
  File "/usr/bin/atomicapp", line 9, in <module>
    load_entry_point('atomicapp==0.1', 'console_scripts', 'atomicapp')()
  File "/usr/lib/python2.7/site-packages/atomicapp/cli/main.py", line 89, in main
    cli.run()
  File "/usr/lib/python2.7/site-packages/atomicapp/cli/main.py", line 70, in run
    args.func(args)
  File "/usr/lib/python2.7/site-packages/atomicapp/cli/main.py", line 25, in cli_stop
    stop.run()
  File "/usr/lib/python2.7/site-packages/atomicapp/run.py", line 187, in run
    self._dispatchGraph()
  File "/usr/lib/python2.7/site-packages/atomicapp/run.py", line 99, in _dispatchGraph
    self._processComponent(component, graph_item)
  File "/usr/lib/python2.7/site-packages/atomicapp/run.py", line 170, in _processComponent
    provider.undeploy()
  File "/usr/lib/python2.7/site-packages/atomicapp/providers/kubernetes.py", line 86, in undeploy
    self._resetReplicas(path)
  File "/usr/lib/python2.7/site-packages/atomicapp/providers/kubernetes.py", line 56, in _resetReplicas
    name = data["id"]
KeyError: 'id'
```

**Comment by navidshaikh** [comment]
Referencing resources in k8s differs from apiVersion `v1beta1` to `v1`.
Latest one uses `metadata.name` field to reference the resources while earlier apiVersion uses `id` field.

Proposing look up in order `metadata.name` and then `id`. The later (`id`) will be looked up in absence of `metadata.name` which means k8s artifacts are of `v1beta1`.

This should also be in sync with support for populating fields (from atomicapp side) for k8s manifests ranging different apiVersions.

@aweiteka @goern @jasonbrooks @vpavlin Thoughts?

**Comment by vpavlin** [comment]
I agree with the proposed order. 

What do you mean by  _in sync with support for populating fields_ ?

**Comment by navidshaikh** [comment]
@vpavlin : 
>What do you mean by in sync with support for populating fields ?

We are populating `self.namespace` from the k8s manifests processed.

For `v1beta1` namespace can be accessed as `namespace` while for `v1` it is accessed as `metadata.namespace`.

So we need to provision populating namespace field based on the apiVersion in the manifest being processed.

On that note, should we migrate completely to `v1` apis for k8s?

@aweiteka @goern @vpavlin : thoughts?

**Comment by goern** [comment]
ja, we should do that, v1 of kubes should be a requirement.

**Comment by jasonbrooks** [comment]
+1 to making kube v1 a requirement

**Comment by navidshaikh** [comment]
> What do you mean by in sync with support for populating fields ?
@vpavlin : I meant about processing `namespace` field from manifest. For `v1beta1` APIs `namespace` field is again at the root level while for `v1` or `v1beta3` the field reside at `metadata.namespace`. However I could not find parsing of manifests for `namespace` field rather we are processing the `namespace` field from config.

**Comment by navidshaikh** [comment]
@vpavlin @goern @aweiteka , others : Can you please review the patch ?

**Comment by LalatenduMohanty** [comment]
@navidshaikh after moving to v1 APIs is it backward compatible to older APIs?

**Comment by navidshaikh** [comment]
> @navidshaikh after moving to v1 APIs is it backward compatible to older APIs?

@LalatenduMohanty : Yes, one can still use manifests of `v1beta1` and the `stop` / resize of app should work. The movement to `v1` API should be separate PR.

**Comment by LalatenduMohanty** [comment]
@navidshaikh thanks. I thought this is the PR to move to V1.

**Comment by navidshaikh** [comment]
should this be closed now?

**Comment by goern** [comment]
which PR closes this? #221 ?

**Comment by navidshaikh** [comment]
Correct! <https://github.com/projectatomic/atomicapp/pull/221>

**Comment by LalatenduMohanty** [comment]
#221 is merged now, lets close this CC  @navidshaikh @goern @vpavlin

**Comment by goern** [comment]
fixed by #221
